internal/dataplane/parser/translators/l4route_atc.go

```diff
@@ -0,0 +1,43 @@
+package translators
+
+import (
+	"github.com/samber/lo"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/kongstate"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/parser/atc"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+)
+
+// ApplyExpressionToL4KongRoute convert route flavor from traditional to expressions
+// against protocols, snis and dest ports.
+func ApplyExpressionToL4KongRoute(r *kongstate.Route) {
+	matchers := []atc.Matcher{}
+
+	protocolMatcher := protocolMatcherFromProtocols(lo.Map(r.Route.Protocols, func(item *string, _ int) string { return *item }))
```

Comment:
There is no need to do this conversion, Kong gateway could convert the `protocol` field in the route and add it to expressions. Also, Kong gateway team suggests us to put protocols in `Protocols` field.